browser/ui/webui/brave_web_ui_controller_factory.cc

```diff
@@ -313,7 +316,11 @@ bool ShouldBlockWalletWebUI(content::BrowserContext* browser_context,
   }
   auto* keyring_service =
       brave_wallet::KeyringServiceFactory::GetServiceForContext(profile);
+#if BUILDFLAG(ENABLE_BRAVE_ANDROID_WEB_WALLET)
+  return keyring_service && !keyring_service->IsWalletCreatedSync();
+#else
   return keyring_service && keyring_service->IsLockedSync();
+#endif
 }
 #endif  // BUILDFLAG(IS_ANDROID)
```

Comment:
1. Should 'block' in case there is no keyring_service
2. Why first case does not block when wallet is locked? Needs a comment at least

Reply:
1. Oh, true! I'm going to fix that.
2. I'm going to add a comment explaining better what's going on in the first case: we extended support to unlock screen _also_ through WebUI screens. Hence, we block only when a Wallet has not been created yet, because we want the user to go through onboarding via native implementation.